pages/pipelines/environment_variables.md

```diff
@@ -255,3 +255,11 @@ environment, and will override any previous values that are set.
 This is the environment your command runs in 🎉
 
 Finally, if your job's commands make any changes to the environment, those changes will only survive as long as the script is running.
+
+### Environment variables in multiple pipeline uploads
+
+In the event that you are uploading multiple pipeline definitions, depending on the order of uploads, environment variable precedence behaves differently.
+
+If you are uploading pipeline definitions concurrently the value of the environment variable will match the uploading pipeline definition.
```

Comment:
Can you please unpack this one a bit more for me? Is it deterministic?

Reply:
@mbelton-buildkite apologies for the delay there. Yes, it's deterministic.

If we use this example, where we upload the pipeline definitions concurrently:
```yaml
steps:
  - command: |
      buildkite-agent pipeline upload .buildkite/p1.yml
      buildkite-agent pipeline upload .buildkite/p2.yml
```
**p1.yml**

```yaml
env:
  FOO: "bar"

steps:
  - label: "P1" 
    command: echo "+++ p1 - $$FOO"
``` 

**p2.yml**
```yaml
env:
  FOO: "baz"

steps:
  - command: echo "+++ p2 - $$FOO"
```

The resulting pipeline will look like this:

![image](https://github.com/buildkite/docs/assets/19416785/905e7d5e-eb98-4540-b0cd-62b8b6f0935b)

The value of `FOO` will match that of the declaration of `FOO` in the respective uploading pipeline YAML.